hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -171,24 +171,32 @@ func (r *NodePoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, err
 	}
 
-	result, err := r.reconcile(ctx, hcluster, nodePool)
-	if err != nil {
-		log.Error(err, "Failed to reconcile NodePool")
-		r.recorder.Eventf(nodePool, corev1.EventTypeWarning, "ReconcileError", "%v", err)
-		if err := patchHelper.Patch(ctx, nodePool); err != nil {
-			log.Error(err, "failed to patch")
-			return ctrl.Result{}, fmt.Errorf("failed to patch: %w", err)
+	result := ctrl.Result{}
+	if isPaused, duration := supportutil.IsReconciliationPaused(log, nodePool.Spec.PausedUntil); isPaused {
+		log.Info("Reconciliation paused", "name", req.NamespacedName, "pausedUntil", *nodePool.Spec.PausedUntil)
+		result = ctrl.Result{RequeueAfter: duration}
+	} else {
```

Comment:
can we avoid having the core default reconciliation in an else? i.e return early above even if that means duplication.

Also the way this is now, it differs from the HC pause logic which reconciles conditions and status even though is paused, we should probably be consistent with that behaviour here.

Reply:
I've modified this so we now reconcile conditions/status, but return before reconciling the desired state